common.mk

```diff
@@ -0,0 +1,149 @@
+	cd $(base_dir) && $(SBT) "runMain example.GenerateSimFiles -td $(build_dir) -sim $(sim_name)"
+
+#########################################################################################
+# create firrtl file rule and variables
+#########################################################################################
+CHISEL_ARGS ?=
+
+$(FIRRTL_FILE) $(ANNO_FILE): $(SCALA_SOURCES) $(sim_dotf)
+	mkdir -p $(build_dir)
+	cd $(base_dir) && $(SBT) "runMain $(PROJECT).Generator $(CHISEL_ARGS) $(build_dir) $(PROJECT) $(MODEL) $(CFG_PROJECT) $(CONFIG)"
+
+#########################################################################################
+# create verilog files rules and variables
+#########################################################################################
+REPL_SEQ_MEM = --repl-seq-mem -c:$(MODEL):-o:$(SMEMS_CONF)
+
+$(VERILOG_FILE) $(SMEMS_CONF): $(FIRRTL_FILE) $(ANNO_FILE) $(TAPEOUT_JAR)
+	$(TAPEOUT) barstools.tapeout.transforms.GenerateTop -o $(VERILOG_FILE) -i $(FIRRTL_FILE) --syn-top $(TOP) --harness-top $(MODEL) -faf $(ANNO_FILE) $(REPL_SEQ_MEM) -td $(build_dir)
+
+$(HARNESS_FILE): $(FIRRTL_FILE) $(ANNO_FILE) $(TAPEOUT_JAR)
+	$(TAPEOUT) barstools.tapeout.transforms.GenerateHarness -o $(HARNESS_FILE) -i $(FIRRTL_FILE) --syn-top $(TOP) --harness-top $(MODEL) -faf $(ANNO_FILE) -td $(build_dir)
+
+# This file is for simulation only. VLSI flows should replace this file with one containing hard SRAMs
+$(SMEMS_FILE): $(SMEMS_CONF) $(MACROCOMPILER_JAR)
+	$(MACROCOMPILER) barstools.macros.MacroCompiler -n $(SMEMS_CONF) -v $(SMEMS_FILE) --mode synflops
+
+#########################################################################################
+# helper rule to just make verilog files
+#########################################################################################
+.PHONY: verilog
+verilog: $(sim_vsrcs)
+
+#########################################################################################
+# run assembly/benchmarks rules
+#########################################################################################
+$(output_dir)/%: $(RISCV)/riscv64-unknown-elf/share/riscv-tests/isa/%
+	mkdir -p $(output_dir)
+	ln -sf $< $@
+
+$(output_dir)/%.run: $(output_dir)/% $(sim)
+	$(sim) +max-cycles=$(timeout_cycles) $< && touch $@
+
+$(output_dir)/%.out: $(output_dir)/% $(sim)
+	$(sim) +verbose +max-cycles=$(timeout_cycles) $< 3>&1 1>&2 2>&3 | spike-dasm > $@
+
+#########################################################################################
+# include build/project specific makefrags made from the generator
+#########################################################################################
+ifneq ($(filter run% %.run %.out %.vpd %.vcd,$(MAKECMDGOALS)),)
+-include $(build_dir)/$(long_name).d
+endif
+
+#########################################################################################
+# default regression tests variables and rules
+# TODO: Remove in favor of each project having its own regression tests?
```

Comment:
The regression tests are meant to be subset of the full asm-tests that a project supports, so I don't think a general-purpose "regression suite" makes sense here.

Also, does anyone even use the regression tests?

Reply:
On `boom`'s end, we do not use the regression tests (we use the full asm and benchmark suite).

Reply:
I use the regression tests when I want to make sure base rocket-chip works correctly. They are much faster than the whole suite. I think we should leave them in for CI at the very least.